packages/patternfly-4/react-core/src/components/Nav/NavItem.js

```diff
@@ -16,50 +16,40 @@ const propTypes = {
   /** Group identifier, will be returned with the onToggle and onSelect callback passed to the Nav component */
   groupId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
   /** Item identifier, will be returned with the onToggle and onSelect callback passed to the Nav component */
-  itemId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+  itemId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  /** If true prevents the default anchor link action to occur. Set to true if you want to handle navigation yourself. */
+  preventDefault: PropTypes.bool,
+  /** Callback for item click */
+  onClick: PropTypes.func
 };
 
 const defaultProps = {
   children: null,
   className: '',
-  to: '#',
+  to: '',
   isActive: false,
   groupId: null,
-  itemId: null
+  itemId: null,
+  preventDefault: false,
+  onClick: null
 };
 
-const NavItem = ({ className, children, to, isActive, groupId, itemId, ...props }) => {
-  const defaultLink = (
+const NavItem = ({ className, children, to, isActive, groupId, itemId, preventDefault, onClick, ...props }) => (
+  <li className={css(styles.navItem, className)} {...props}>
     <NavContext.Consumer>
       {context => (
         <a
           href={to}
-          onClick={e => context.onSelect(e, groupId, itemId)}
+          onClick={e => context.onSelect(e, groupId, itemId, to, preventDefault, onClick)}
           className={css(styles.navLink, isActive && styles.modifiers.current, className)}
           aria-current={isActive ? 'page' : null}
         >
           {children}
         </a>
       )}
     </NavContext.Consumer>
-  );
-  const reactElement = React.isValidElement(children);
-  const clonedChild = (
-    <NavContext.Consumer>
-      {context =>
-        React.cloneElement(children, {
-          onClick: e => context.onSelect(e, groupId, itemId),
-          className: css(styles.navLink, isActive && styles.modifiers.current, className)
-        })
-      }
-    </NavContext.Consumer>
-  );
-  return (
-    <li className={css(styles.navItem, className)} {...props}>
-      {reactElement ? clonedChild : defaultLink}
-    </li>
-  );
-};
+  </li>
+);
```

Comment:
this bit was useful if we want to allow the consumer to still render a `Link` instead of just the `NavItem`, i.e.:

```
      <NavItem isActive={isActive}>
        <Link
          id={id}
          to={this.to}
          target={target}
          onClick={onClick}
          className={classNames({
            'pf-c-nav__link': true,
            'pf-m-current': isActive,
            'pf4-co-external-link': isExternal
          })}
        >
          {name}
```

I find it preferable to use that approach as well (it's a bit more explicit the render the Link tag/anchor separately).

Reply:
(in that case, the NavItem just renders the `li`)

Reply:
Thanks, I've added it back in